### PR TITLE
Add Unregister method

### PR DESCRIPTION
**Pull request by aureolebigben**
I did some modifications to add Unregister method

### CODE REVIEW

**Comment by aureolebigben** [comment]
Don't know why build didn't pass on your travis CI. It works fine with mine.

File: README.md [hunk 1]
```diff
@@ -1,9 +1,11 @@
 # About
 
-[![Build Status](https://travis-ci.org/tebeka/atexit.svg?branch=master)](https://travis-ci.org/tebeka/atexit)
+[![Build Status](https://travis-ci.org/aureolebigben/atexit.svg?branch=master)](https://travis-ci.org/aureolebigben/atexit)
```
Comment (tebeka) [inline README.md]: Change back to `tebeka`

File: README.md [hunk 2]
```diff
@@ -15,25 +17,31 @@ package main
 import (
     "fmt"
 
-    "github.com/tebeka/atexit"
+    "github.com/aureolebigben/atexit"
```
Comment (tebeka) [inline README.md]: Change back to `tebeka`

File: README.md [hunk 3]
```diff
@@ -15,25 +17,31 @@ package main
...
 )
 
 func handler() {
     fmt.Println("Exiting")
 }
 
+func fooHandler() {
```
Comment (tebeka) [inline README.md]: I general I don't like the `foo` and `bar` names. Try to think is a better name - maybe `unregistredHandler`?

File: README.md [hunk 4]
```diff
@@ -15,25 +17,31 @@ package main
...
+    fmt.Println("not called")
+}
+
 func main() {
-	atexit.Register(handler)
-	atexit.Exit(0)
+    _ = atexit.Register(handler)
```
Comment (tebeka) [inline README.md]: What is the value of `_ =`? How does it contribute?

File: README.md [hunk 5]
```diff
@@ -15,25 +17,31 @@ package main
...
+    fooId := atexit.Register(fooHandler)
+    atexit.Unregister(fooId)
+    atexit.Exit(0)
 }
 ```
 
 # Install
 
-    go get github.com/tebeka/atexit
+    go get github.com/aureolebigben/atexit
```
Comment (tebeka) [inline README.md]: Revert to `tebeka`

File: atexit.go [hunk 6]
```diff
@@ -35,17 +35,28 @@ import (
...
 	// Version is package version
-	Version = "0.2.0"
+	Version = "0.3.0"
 )
 
-var handlers = []func(){}
+type HandlerId uint
+
+func (h *HandlerId) Get() HandlerId {
+	defer func() {
```
Comment (tebeka) [inline atexit.go]: This is a nice trick but too clever to my taste. Go is about simplicity and I prefer it here as well. Have a global `var nextID HandlerID` and incement it.

Also, use the lock (or sync/atomic)

File: atexit.go [hunk 7]
```diff
@@ -35,17 +35,28 @@ import (
...
+var handlers = map[HandlerId]func(){}
 var handlersLock sync.RWMutex
 var once sync.Once
+var nextFuncId HandlerId
 
 func runHandler(handler func()) {
 	defer func() {
 		if err := recover(); err != nil {
-			fmt.Fprintln(os.Stderr, "error: atexit handler error:", err)
+			_, _ = fmt.Fprintln(os.Stderr, "error: atexit handler error:", err)
```
Comment (tebeka) [inline atexit.go]: What's the value of `_, _ =`?

File: atexit_test.go [hunk 8]
```diff
@@ -16,19 +16,28 @@ func TestRegister(t *testing.T) {
...
+}
+
 func TestHandler(t *testing.T) {
 	err := exec.Command("go", "install").Run()
 	if err != nil {
 		t.Fatalf("can't install - %s", err)
 	}
 
-	gofile := "/tmp/atexit-testprog.go"
+	gofile := "./testprog/atexit-testprog.go"
```
Comment (tebeka) [inline atexit_test.go]: How is `testprog` better than `/tmp`? Windows? I prefer to use `ioutil.TempDir` in this case

File: go.mod [hunk 9]
```diff
@@ -1 +0,0 @@
-module github.com/tebeka/atexit
```
Comment (tebeka) [inline go.mod]: Why was this deleted?

**Review by tebeka — CHANGES_REQUESTED** [review]
Thanks! I'd love if you'll address my comments.
But before you do that - what is the need for `Unregister`? Why did you write it?

File: README.md (hunk 3 above)
Comment (aureolebigben) [reply README.md]: Think unregistredHandler is a good name, i use fooHandler because I had no inspiration

File: README.md (hunk 4 above)
Comment (aureolebigben) [reply README.md]: i added `_ =` because it's a best practice. It's use to say that we don't want the returned value

File: atexit.go (hunk 7 above)
Comment (aureolebigben) [reply atexit.go]: fmt.Fprintln return `n` and `err` that's why i use `_, _ =` to say explicitly that we want to ignore returned values

File: atexit_test.go (hunk 8 above)
Comment (aureolebigben) [reply atexit_test.go]: That's just because whith /tmp my Travis CI don't want to build. So your free to use what you want. Don't why build pass on my Travis and not your's and vice versa

File: go.mod (hunk 9 above)
Comment (aureolebigben) [reply go.mod]: Ignore this, it's just an error sorry

**Comment by aureolebigben** [comment]
> Thanks! I'd love if you'll address my comments.
> But before you do that - what is the need for `Unregister`? Why did you write it?

In one of my application i have some case where program can exit at a time where i want to execute some handlers. 
But sometime certains handlers don't have to be executed so i added this method.

It's just my idea but i think that it's a good practice to allow people to unregister something that they registered.

Sorry if my english is not very good, I'm french x)

File: atexit.go (hunk 6 above)
Comment (aureolebigben) [reply atexit.go]: Feel free to modify the code.
I'll be happy to see how you imagine this.

Comment (tebeka) [reply atexit.go]: See https://github.com/tebeka/atexit/pull/4

**Comment by tebeka** [comment]
Thanks for the idea. I've implemented it a bit differently in #4
